Allow review saves with incomplete fields

benjaminJohnson2204 opened the pull request:
- Allow reviewers to save reviews with incomplete fields; only the fields they enter a score for will be saved on the `review.fields`
- Update logic for checking if a review is complete to consider 3 statuses: not started (no fields are filled in), in progress (some but not all fields are filled in), and completed (all fields are filled in), checking against the fields on the review's stage
- Your reviews table on homepage now displays status for each review, sorting by not started, then in progress, then completed
- Candidates by stage table now only shows scores & average score for complete reviews
- Updated next review endpoint to get a not started review, if any, falling back to an in progress review
- Removed support for boolean review fields because 1. we're not using them, and 2. it would be complicated to figure out if an unchecked checkbox means the field should be false, or just isn't filled in yet. We'd have to figure that out if we ever need boolean fields again in the future - maybe have a select menu with options for "Yes", "No", and "Please select a value" which indicates nothing selected

Tested these changes by clearing/setting review fields and checking the review status afterward

<img width="1848" height="197" alt="Screenshot from 2025-08-24 13-01-45" src="https://github.com/user-attachments/assets/9bb0c425-e85e-4dbc-bfc0-8df977feacce" />

yxli001 reviewed (commented):
looks good! should we only show the popup to go to the next review if the review was completed? maybe if the review is uncompleted when saved, just show an alert saying saved successfully

also unrelated to this pr but i noticed the starting quarters were messed up on your homepage, is that because you have legacy data in your local db?

benjaminJohnson2204 commented:
Yeah good point, especially because if a reviewer is doing their last review and they save it as a draft, then the next review endpoint would return that same review since it's the only incomplete one, which would be a weird UX lol

I updated the frontend to just show a "Saved as draft" alert if not all the fields are complete. Also fixed a few bugs with determining review status

Yeah, the quarters are messed up because the application is from when fall quarter mapped to 3 instead of 2

<img width="1060" height="1020" alt="Screenshot from 2025-08-26 20-09-13" src="https://github.com/user-attachments/assets/ccd25d47-1936-4b0f-8aa7-3c856f29d2b9" />

yxli001 reviewed (approved):
LGTM! thanks for making that change
